web/lib/supabase/stuff.ts

```diff
@@ -0,0 +1,19 @@
+import { createClient } from '@supabase/supabase-js'
+import { DEV_CONFIG } from 'common/envs/dev'
+
+// TODO: rewrite entirely. Set up types, put keys as env vars or set up row level access
+
+const SUPABASE_READ_KEY =
+  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1mb2RvbnpueWZ4bGxjZXp1ZmdyIiwicm9sZSI6ImFub24iLCJpYXQiOjE2Njc5ODgxNjcsImV4cCI6MTk4MzU2NDE2N30.RK8CA3G2_yccgiIFoxzweEuJ2XU5SoB7x7wBzMKitvo'
+
+const supabase = createClient(DEV_CONFIG.supabaseUrl ?? '', SUPABASE_READ_KEY)
+
+export async function searchUsers(prompt: string) {
+  const { data } = await supabase
+    .from('users')
+    .select('data')
+    .or(`data->>username.ilike.%${prompt}%,data->>name.ilike.%${prompt}%`)
```

Comment:
FWIW, since we only have like 10k users this will probably be fine, but you should get in the habit of adding appropriate SQL indexes to serve queries like this.